### DISCUSSION TITLE
Can I push Docker images to Nexus?

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#285] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11

### Description

Let's say we want to accelerate the debugging of an issue in a non-production environment, and we want to build an image locally, skipping all the tests so that we can quickly apply changes to see if we can solve a problem, is there a way to push an image that was built locally to our Nexus?

Because [ there's a way ](https://help.sonatype.com/repomanager3/nexus-repository-administration/formats/docker-registry/pushing-images), you just have to build the image locally with the same full name (registry host + image + tag) of the image.

For example: 

```sh
docker push nexus.mia-platform.eu/<path>/helloworld-service:latest
``` 

However, even though I logged in via `docker login nexus.mia-platform.eu` I get:

`unauthorized: access to the requested resource is not authorized`

The question therefore is: do I get this errore because I don't have access to the specific asset, or developers, in general, cannot push images to Nexus?

It'd be helpful if at least the TL could push images in "break the glass situations".

Thank you,

Luca

**Comment by fredmaggiowski** [comment, marked as answer]
Hey, I think this is a safety measure from the PaaS infrastructure team; maybe you should discuss this matter with them or open a service request.

IMHO this is an absolute bad practice we must avoid, so I'd agree with them not granting access to the registry but forcing image push from a versioned repository rather than from manual actions; as this behavior would lead to inconsistent image management as there would be unpredictable changes to images 🤷

**Reply by theliuk** [reply]
I see, but there are cases in which you cannot spend time waiting for CI pipelines to build your image; I'm saying that we should be able to change manually the images that are produced by the CI, but there's should be at least the possibility to push an image in a "personal space".
This should not be the routine, but the exception (and not for production images).

Luca

**Comment by frengooo** [comment]
I remember too it's not allowed by the PaaS infrastructure team as a security measure.

Can't you use telepresence through the "Debug" section of the console?

**Reply by theliuk** [reply]
Nope, not from Mac OS, 'cause the only version of telepresence that Mac OS supports is not supported by our cluster (this was explicitly requested by @giulioroggero ).

**Reply by frengooo** [reply]
Are you sure? We recently tried Telepresence v2 on a client's cluster and it worked.

**Reply by theliuk** [reply]
Wow, seriously? At the time I was told that telepresence v2 wouldn't be supported by our clusters. Good that we changed our mind!
I'll try,

Thank you,

Luca
